packages/connection-manager/src/js/Connection/XHRConnection.js

```diff
@@ -0,0 +1,198 @@
+import AbstractConnection from "./AbstractConnection";
+import {
+  CONNECTION_EVENT_CLOSE,
+  CONNECTION_EVENT_OPEN,
+  CONNECTION_EVENT_PROGRESS,
+  CONNECTION_EVENT_ABORT,
+  CONNECTION_EVENT_ERROR,
+  CONNECTION_EVENT_SUCCESS,
+  CONNECTION_STATE_CANCELED,
+  CONNECTION_STATE_INIT,
+  CONNECTION_STATE_DONE,
+  CONNECTION_STATE_STARTED
+} from "./ConnectionConstants";
+
+export const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {string} [_method=GET] – used method
+   * @param {mixed} [_data] – payload for request
+   * @param {string} [_header] – additional headers (like content-type)
+   */
+  constructor(url, _method = "GET", _data = null, _header = null) {
```

Comment:
Let's pass all of the optional arguments as an options object as suggested in   https://github.com/dcos/dcos-ui/pull/2409#discussion_r138210249, to align the interface with known request libs as well as the coming [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request/Request ) (fetch) in the standard lib.

## XHRConnection

```js
var myRequest = new XHRConnection(url, options);

```` 

### Parameter

* `url` - Defines the resource that you wish to request
* `options` [optional] - Optional options object containing any custom settings that you want to apply to the request.

#### Options

* `method` - Request method, e.g., GET, POST.
* `headers` -  Any headers you want to add to your request, contained within a Headers object or an object literal with ByteString values.
* body: Any body that you want to add to your request: this can be a Blob, BufferSource, FormData, URLSearchParams, or USVString object. Note that a request using the GET or HEAD method cannot have a body.

What do you think?

/cc @dcos/frontend-foundation